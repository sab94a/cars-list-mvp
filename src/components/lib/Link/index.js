// @flow

import React from 'react';
import cx from 'classnames';
import type { AbstractComponent } from 'react';
import type { Link as Props } from 'components/lib/types';
import styles from './index.module.scss';

class Link extends React.Component<Props> {
    static Component: AbstractComponent<*> | string = 'a';

    render() {
        const { href, children, className, ...rest } = this.props;

        return (
            <Link.Component className={cx( className, styles.root )} { ...rest } href={ href }>{ children }</Link.Component>
        );
    }
}

export default Link;
