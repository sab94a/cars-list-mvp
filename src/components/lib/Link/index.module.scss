@import 'components/lib/variables.scss';

.root {
    color: $primary;

    &:hover, &:focus {
        text-decoration: underline;
        color: $primary-pressed;    
    }
}
