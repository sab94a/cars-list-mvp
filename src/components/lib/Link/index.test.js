import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from './'

describe('<Link />', () => {
    const href = '/path';
    const title = 'title'
    let component;

    const create = ({title, ...params}) => {
        component = mount(<Link { ...params }>{ title }</Link>);
    };

    it('Should render correctly', () => {
        create({ href, title });

        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should have right title', () => {
        create({ href, title });

        expect(component.text()).toEqual(title);
    });

    it('Should have right link', () => {
        create({ href, title });

        expect(component.find('a').prop('href')).toEqual(href);
    });

    it('Should pass extra params', () => {
        const className = 'test';
        create({ href, title, className });

        expect(component.find('a').prop('className')).toEqual(className);
    });
});
