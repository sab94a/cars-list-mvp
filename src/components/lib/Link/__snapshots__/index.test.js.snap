// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Should render correctly 1`] = `
<Link
  href="/path"
>
  <a
    className="root"
    href="/path"
  >
    title
  </a>
</Link>
`;
